src/bin/pgcopydb/ld_stream.h

```diff
@@ -309,6 +345,9 @@ typedef struct StreamContext
 	/* transform needs some catalog lookups (pkey, type oid) */
 	DatabaseCatalog *sourceDB;
 
+	/* hash table acts as a cache for tables with generated columns */
+	GeneratedColumnsCache *generatedColumnsCache;
```

Comment:
Why is this a pointer? Is this an array? a linked-list? a single item that is malloced later?

Reply:
@dimitri, it is a hash table as explained in the comment above.

Reply:
Yeah but a hash-table over what? We might have more than one column per table, and we have a single pointer. What do we point to? Certainly the hash-table has one entry per column name, and also points to somewhere... where does it point to?

Reply:
Same as the other comment, there is never a need for a sequential access to the data allocated, so all we have to access the malloc'ed areas is the hash-table.